WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/data/watchface/ColorStyleIdAndResourceIds.kt

```diff
@@ -0,0 +1,157 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.alpha.data.watchface
+
+import android.content.Context
+import android.graphics.drawable.Icon
+import androidx.wear.watchface.style.UserStyleSetting
+import androidx.wear.watchface.style.UserStyleSetting.ListUserStyleSetting
+import com.example.android.wearable.alpha.R
+
+// Defaults for all styles.
+// X_COLOR_STYLE_ID - id in watch face database for each style id.
+// X_COLOR_STYLE_NAME_RESOURCE_ID - String name to display in the user settings UI for the style.
+// X_COLOR_STYLE_ICON_ID - Icon to display in the user settings UI for the style.
+const val AMBIENT_COLOR_STYLE_ID = "ambient_style_id"
+private const val AMBIENT_COLOR_STYLE_NAME_RESOURCE_ID = R.string.ambient_style_name
+private const val AMBIENT_COLOR_STYLE_ICON_ID = R.drawable.white_style
+
+const val RED_COLOR_STYLE_ID = "red_style_id"
+private const val RED_COLOR_STYLE_NAME_RESOURCE_ID = R.string.red_style_name
+private const val RED_COLOR_STYLE_ICON_ID = R.drawable.red_style
+
+const val GREEN_COLOR_STYLE_ID = "green_style_id"
+private const val GREEN_COLOR_STYLE_NAME_RESOURCE_ID = R.string.green_style_name
+private const val GREEN_COLOR_STYLE_ICON_ID = R.drawable.green_style
+
+const val BLUE_COLOR_STYLE_ID = "blue_style_id"
+private const val BLUE_COLOR_STYLE_NAME_RESOURCE_ID = R.string.blue_style_name
+private const val BLUE_COLOR_STYLE_ICON_ID = R.drawable.blue_style
+
+const val WHITE_COLOR_STYLE_ID = "white_style_id"
+private const val WHITE_COLOR_STYLE_NAME_RESOURCE_ID = R.string.white_style_name
+private const val WHITE_COLOR_STYLE_ICON_ID = R.drawable.white_style
+
+/**
+ * Represents watch face color style options the user can select (includes the unique id, the
+ * complication style resource id, and general watch face color style resource ids).
+ *
+ * The companion object offers helper functions to translate a unique string id to the correct enum
+ * and convert all the resource ids to their correct resources (with the Context passed in). The
+ * renderer will use these resources to render the actual colors and ComplicationDrawables of the
+ * watch face.
+ */
+enum class ColorStyleIdAndResourceIds(
+    val id: String,
+    val nameResourceId: Int,
+    val iconResourceId: Int,
+    val complicationStyleDrawableId: Int,
+    val primaryColorId: Int,
+    val secondaryColorId: Int,
+    val backgroundColorId: Int,
+    val outerElementColorId: Int
```

Comment:
Nit: Could we add `@StringRes`, `@DrawableRes`, etc.?

Reply:
done.